.github/workflows/demo-testing.yml

```diff
@@ -9,8 +9,224 @@ on:
+              - 'external-providers/yq-external-provider/**'
+            c-sharp:
+              - '**'
+              - '!external-providers/**'
+
+  build-bases:
+    needs: detect-changes
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - provider: golang-dep-provider
+            image_name: golang-dep-provider
+            dockerfile: external-providers/golang-dependency-provider/Dockerfile
+          - provider: analyzer-lsp
+            image_name: konveyor-analyzer-lsp
+            dockerfile: Dockerfile
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build ${{ matrix.provider }} image
+        uses: shawn-hurley/ci/build-image@main
+        with:
+          repo: konveyor/analyzer-lsp
+          checked_out: true
+          image_name: ${{ matrix.image_name }}
+          image_tag: ${{ needs.detect-changes.outputs.image_tag }}
+          dockerfile_path: ${{ matrix.dockerfile }}
+          build_context: .
+
+  build-all-providers:
+    needs: [detect-changes, build-bases]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - provider: java
+            image_name: java-provider
+            dockerfile: external-providers/java-external-provider/Dockerfile
+            build_args: JAVA_BUNDLE_TAG=latest
+            checked_out: true
+          - provider: generic
+            image_name: generic-provider
+            dockerfile: external-providers/generic-external-provider/Dockerfile
+            needs_golang_dep: true
+            build_args: GOLANG_DEP_IMAGE=localhost/golang-dep-provider:${{ needs.detect-changes.outputs.image_tag }}
+            checked_out: true
+          - provider: yq
+            image_name: yq-provider
+            dockerfile: external-providers/yq-external-provider/Dockerfile
+            checked_out: true
+          - provider: c-sharp
+            image_name: c-sharp-provider
+            dockerfile: Dockerfile
+            repo: konveyor/c-sharp-analyzer-provider
+            checked_out: false
+    steps:
+      - uses: actions/checkout@v3
+        if: needs.detect-changes.outputs[matrix.provider] == 'true'
+
+      - name: Build ${{ matrix.provider }} provider image
+        if: needs.detect-changes.outputs[matrix.provider] == 'true'
+        uses: shawn-hurley/ci/build-image@main
+        with:
+          repo: ${{ matrix.repo || 'konveyor/analyzer-lsp' }}
+          checked_out: ${{ matrix.checked_out }}
+          image_name: ${{ matrix.image_name }}
+          image_tag: ${{ needs.detect-changes.outputs.image_tag }}
+          dockerfile_path: ${{ matrix.dockerfile }}
+          build_context: .
+          base_image: ${{ matrix.needs_golang_dep == true && format('golang-dep-provider--{0}', needs.detect-changes.outputs.image_tag) || '' }}
+          build_args: ${{ matrix.build_args || '' }}
+
+  provider-tests:
+    needs: [detect-changes, build-all-providers]
```

Comment:
Just to be sure, we build all providers all time, so the speed-up is exected to be introduced with running providers tests in parallel matrix below, right?